insalan/mailer.py

```diff
@@ -0,0 +1,255 @@
+        )
+        email.attach(
+            TicketManager.create_pdf_name(ticket),
+            ticket_pdf,
+            "application/pdf"
+        )
+
+        if self.TEST:
+            email.send()
+        else:
+            self.queue.append(email)
+
+    def send_tournament_mail(self, user_object: User, title: str, content: str, attachment: str):
+        """
+        Send a mail
+        """
+        connection = get_connection(
+            fail_silently=False,
+            username=self.MAIL_FROM,
+            password=self.MAIL_PASS,
+            host=self.MAIL_HOST,
+            port=self.MAIL_PORT,
+            use_ssl=self.MAIL_SSL,
+        )
+        email = EmailMessage(
+            insalan.settings.EMAIL_SUBJECT_PREFIX + title,
+            content,
+            self.MAIL_FROM,
+            [user_object.email],
+            connection=connection,
+        )
+        if attachment:
+            email.attach(attachment.name, attachment.read())
+
+        if self.TEST:
+            email.send()
+        else:
+            self.queue.append(email)
+
+    def send_first_mail(self):
+        """
+        Send the first mail in the queue.
+        """
+        if len(self.queue) == 0:
+            return
+        self.queue[0].send()
+        self.queue.pop(0)
+
+class MailManager:
+    """
+    Manage emails.
+    """
+    mailers = {}
+
+    @staticmethod
+    def get_mailer(MAIL_FROM: str) -> UserMailer:
+        """
+        Get a mailer for a specific email address.
+        """
+        if MAIL_FROM not in MailManager.mailers:
+            return None
+        return MailManager.mailers[MAIL_FROM]
+    
+    @staticmethod
+    def get_default_mailer() -> UserMailer:
+        """
+        Get a mailer for a specific email address.
+        """
+        if len(MailManager.mailers) == 0:
+            return None
+        return list(MailManager.mailers.values())[0]
+
+    @staticmethod
+    def add_mailer(MAIL_HOST:str, MAIL_PORT: str, MAIL_FROM: str, MAIL_PASS: str, MAIL_SSL:bool, TEST: bool = False):
+        """
+        Add a mailer for a specific email address.
+        """
+        MailManager.mailers[MAIL_FROM] = UserMailer(MAIL_HOST, MAIL_PORT, MAIL_FROM, MAIL_PASS, MAIL_SSL, TEST=TEST)
+
+    @staticmethod
+    def send_queued_mail():
+        """
+        Send first mail in all mailer's queue.
+        """
+        for mailer in MailManager.mailers.values():
+            mailer.send_first_mail()
+
+def start_scheduler():
+    # Remove apscheduler logs
+    logging.getLogger('apscheduler.executors.default').setLevel(logging.WARNING)
+
+    # Check if we are in test mode
+    TEST = 'test' in sys.argv
+    print(TEST, file=sys.stderr)
+
+    # Add mailers
+    for auth in insalan.settings.EMAIL_AUTH:
+        mailer = insalan.settings.EMAIL_AUTH[auth]
+        MailManager.add_mailer(mailer["host"], mailer["port"], mailer["from"], mailer["pass"], mailer["ssl"], TEST=TEST)
+
+    # Start scheduler
+    if not TEST:
```

Comment:
Just in case, you may want to add that the mailer is disabled in test mode (in `settings.py`, above the mailer config string). Some poor soul might find themself trying to test it in the future and wonder why it's not doing anything.